src/appleseed.python/bindblendermeshconverter.cpp

```diff
@@ -118,78 +113,119 @@ void convert_bl_mesh(
         }
     }
 
-    int uv_index = 0;
-
     // Push triangles
-    for (size_t tri_index = 0; tri_index < triangle_length; ++tri_index)
+    for (size_t tri_index = 0; tri_index < bl_faces_size; ++tri_index)
     {
-        MeshFace &face = triangles[tri_index];
+        const MeshFace &face = bl_faces[tri_index];
         blender_mesh->push_triangle(
             Triangle(
                 face.v[0],
                 face.v[1],
                 face.v[2],
                 face.mat_nr));
+        if (face.v[3] != 0)
+        {
+            blender_mesh->push_triangle(
+                Triangle(
+                    face.v[2],
+                    face.v[3],
+                    face.v[0],
+                    face.mat_nr));
+        }
     }
 
+    
     // Tie vertex normals to mesh faces
-    if (export_normals == true)
+    if (export_normals)
     {
-        for (size_t tri_index = 0; tri_index < triangle_length; ++tri_index)
+        size_t bl_face_index = 0;
+        for (size_t tri_index = 0; tri_index < bl_faces_size; ++tri_index, ++bl_face_index)
         {
-            MeshFace &face = triangles[tri_index];
-            Triangle& tri = blender_mesh->get_triangle(tri_index);
+            const MeshFace& face = bl_faces[tri_index];
```

Comment:
Is it intentional that you're using `tri_index` to index into _faces_, and `bl_face_index` to index into triangles?